makiflow/layers/neural_texture.py

```diff
@@ -46,15 +48,16 @@ def __init__(self, width, height, num_f, name, text_init=None):
                          named_params_dict=named_params_dict
         )
 
-    def _forward(self, x):
-        # Normalize the input UV map so that its coordinates are within [-1, 1] range.
-        x = x * 2.0 - 1.0
-        batch_size = x.get_shape().as_list()[0]
-        expanded_texture = tf.concat([self._texture] * batch_size, axis=0)
-        return grid_sample(expanded_texture, x)
+    def _forward(self, X, type_graph_operation=MakiRestorable.TEST_PREFIX):
+        with tf.name_scope(type_graph_operation + super().get_name()):
```

Comment:
Why call `super().get_name()` and not `self.get_name()`? The name attribute is directly related to the instance this class, not the parent one. It may be necessary to discuss what to call from `super` and `self`.